implementation/src/main/java/io/smallrye/config/ConfigValueConfigSource.java

```diff
@@ -8,9 +8,30 @@
 
 import org.eclipse.microprofile.config.spi.ConfigSource;
 
+/**
+ * Extends the original {@link ConfigSource} to expose methods that return a {@link ConfigValue}. The
+ * {@link ConfigValue} allows to retrieve additional metadata associated with the configuration resolution.
+ *
+ * This is to workaround the limitation that the original {@link ConfigSource} exposes everything as plain Strings and
+ * is not possible to retrieve additional information associated with the Configuration. The ConfigValueConfigSource
+ * tries to make this possible.
+ *
+ * Ideally, this should move the the MicroProfile Config API, once the concept is well-proven.
+ */
 public interface ConfigValueConfigSource extends ConfigSource {
+    /**
+     * Return the {@link ConfigValue} for the specified property in this configuration source.
+     *
+     * @param propertyName the property name
+     * @return the ConfigValue, or {@code null} if the property is not present
+     */
     ConfigValue getConfigValue(String propertyName);
 
+    /**
+     * Return the properties in this configuration source as a Map of String and ConfigValue.
+     *
+     * @return a map containing properties of this configuration source
+     */
     Map<String, ConfigValue> getConfigValueProperties();
 
     @Override
```

Comment:
There probably should be some JavaDocs explaining how these default methods are implemented.